### PR TITLE
Fixed broken SpatialTransformerNetwork example

### PR DESCRIPTION
**Pull request by facundoq**
Re-added option to include a SpatialTransformerNetwork layer in the lenet/mnist example, and a cmd argument in train_mnist to activate it.

The symbol "lenet" had a parameter to add a STN layer, but it was broken because the definition of the location network was missing (got removed in a previous commit, i'm guessing accidentally).

I also added a command line argument to train_mnist.py to add the STN layer to the network.

### CODE REVIEW

**Comment by sxjscience** [comment]
In fact, the get_loc() symbol is not correct in the previous STN example. I'll PR a correct one very soon.

**Comment by facundoq** [comment]
@sxjscience The original STN paper uses different localisation networks for each dataset. I don't see how the current localisation network can be _wrong_ since it basically gives the same results as lenet for vanilla mnist. 

If you want to use STNs with cluttered MNIST, then sure, you'll probably need/want another type of localisation network. 

However that doesn't change the fact that the current lenet-stn example code is a bit broken because (1) there are no command line arguments to set the add_stn parameter to True (2) if add_stn is set to True there is no get_loc function to create a localisation network and you'll get an error.

If having a STN in vanilla MNIST seems overkill because it doesn't give any apparent performance advantages and you want to remove the example altogether and add another example with cluttered MNIST that's ok, but I think that the example is useful because it shows how to add a STN layer with a simple model and dataset.

File: example/image-classification/symbols/lenet.py [hunk 1]
```diff
@@ -5,6 +5,22 @@
...
+    """
+    the localisation network in lenet-stn, it will increase acc about more than 1%,
+    when num-epoch >=15
+    """
+    loc = mx.symbol.Convolution(data=data, num_filter=30, kernel=(5, 5), stride=(2,2))
+    loc = mx.symbol.Activation(data = loc, act_type='relu')
+    loc = mx.symbol.Pooling(data=loc, kernel=(2, 2), stride=(2, 2), pool_type='max')
+    loc = mx.symbol.Convolution(data=loc, num_filter=60, kernel=(3, 3), stride=(1,1), pad=(1, 1))
+    loc = mx.symbol.Activation(data = loc, act_type='relu')
+    loc = mx.symbol.Pooling(data=loc, global_pool=True, kernel=(2, 2), pool_type='avg')
```
Comment (sxjscience) [inline example/image-classification/symbols/lenet.py]: We can keep the example but we need to remove the kernel=(2,2) here

My point is that we cannot understand STN by running this example. We need a more accurate example that compare networks with/without STN.

**Comment by facundoq** [comment]
I think there are two types of "understanding" you may want to accommodate:
1) How to use the API. For this you need the example to be as simple as
possible (even if it doesn't make much sense in terms of complexity vs
accuracy, as in the example in this pr)
2) How to effectively use STNs: For this, sure, you may want to change the
localisation network, but most important you'll need another dataset
(vanilla mnist is too easy) .

**Comment by piiswrong** [comment]
what's the decision on this?

**Comment by sxjscience** [comment]
We can merge this first and I will PR a complete example.

**Comment by mli** [comment]
@sxjscience is it ready to merge?

I'm going to first close the PR since there is no activity for more than 2 months. But please feel free to re-open it and update with the master.

**Comment by sxjscience** [comment]
It's ready to merge.

**Pull request merged**